server/service/src/invoice/common.rs

```diff
@@ -14,3 +18,39 @@ pub fn generate_invoice_user_id_update(
         })
     })
 }
+
+#[derive(Debug, PartialEq)]
+pub struct AddToShipmentFromMasterListInput {
+    pub shipment_id: String,
+    pub master_list_id: String,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum AddToShipmentFromMasterListError {
+    ShipmentDoesNotExist,
+    NotThisStoreShipment,
+    CannotEditShipment,
+    MasterListNotFoundForThisName,
+    NotAnOutboundShipment,
+    NotAnInboundShipment,
```

Comment:
Hmm, I think the types should be seperate (even though they are pretty much mirror image). When we do mapping from service error to graphql error, we want to do exhaustive match for errors that are relevant to that particular service. So for say inbound shipment, doing a match on NonAnOutboundShipment would make little sense, and doing a wild mark match would mean compiler won't help us when adding new error.